hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java

```diff
@@ -85,6 +85,9 @@ public String toString() {
      * Executes all the operations on the partitions.
      */
     <T> Map<Integer, T> invoke() throws Exception {
+        if (Thread.currentThread().getName().contains(".async.thread-")) {
```

Comment:
Logic based on a name of a thread is fragile.

Reply:
@sancar added a new interface some time ago. I think it was called UnblockableThread or so.

Reply:
Yeah, this was just to try the tests. Can you show me how to check better? I was unable to figure out.